registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/PacketManagerService.java

```diff
@@ -1,277 +1,306 @@
+
+        if (response.getErrors() != null && response.getErrors().size() > 0) {
+            regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), id, JsonUtils.javaObjectToJsonString(response));
+            throw new PacketManagerException(response.getErrors().get(0).getErrorCode(), response.getErrors().get(0).getMessage());
+        }
+
+        InfoResponseDto infoResponseDto = objectMapper.readValue(JsonUtils.javaObjectToJsonString(response.getResponse()), InfoResponseDto.class);
+
+        return infoResponseDto;
+    }
+
+    public void addOrUpdateTags(String id, Map<String, String> tags) throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+        UpdateTagRequestDto updateTagRequestDto = new UpdateTagRequestDto(id, tags);
+
+        RequestWrapper<UpdateTagRequestDto> request = new RequestWrapper<>();
+        request.setId(ID);
+        request.setVersion(VERSION);
+        request.setRequesttime(DateUtils.getUTCCurrentDateTime());
+        request.setRequest(updateTagRequestDto);
+        ResponseWrapper<Void> response = (ResponseWrapper) restApi.postApi(ApiName.PACKETMANAGER_UPDATE_TAGS, "", "", request, ResponseWrapper.class);
+
+        if (response.getErrors() != null && response.getErrors().size() > 0) {
+            regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), id, JsonUtils.javaObjectToJsonString(response));
+            throw new PacketManagerException(response.getErrors().get(0).getErrorCode(), response.getErrors().get(0).getMessage());
+        }
+    }
+
+	@SuppressWarnings("unchecked")
+	public void deleteTags(String id, List<String> tags)
+			throws ApisResourceAccessException, PacketManagerException, JsonProcessingException {
+		DeleteTagRequestDTO deleteTagREquestDto = new DeleteTagRequestDTO(id, tags);
+		RequestWrapper<DeleteTagRequestDTO> request = new RequestWrapper<>();
+		request.setId(ID);
+		request.setVersion(VERSION);
+		request.setRequesttime(DateUtils.getUTCCurrentDateTime());
+		request.setRequest(deleteTagREquestDto);
+		ResponseWrapper<DeleteTagResponseDTO> response = (ResponseWrapper<DeleteTagResponseDTO>) restApi
+				.postApi(ApiName.PACKETMANAGER_DELETE_TAGS, "", "",
+				request, ResponseWrapper.class);
+
+		if (response.getErrors() != null && response.getErrors().size() > 0) {
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					id, JsonUtils.javaObjectToJsonString(response));
+			throw new PacketManagerException(response.getErrors().get(0).getErrorCode(),
+					response.getErrors().get(0).getMessage());
+		}
+
+
+	}
+
+    public Map<String, String> getTags(String id, List<String> tagNames) throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
```

Comment:
can we have one more method called getAllTags() and internally call this method with empty array ?

Reply:
I am not using this method in MosipVerticleManager.java. I am separately calling tags api in MosipVerticleManager because its in core. But we have PacketManagerService.java in info-storage-service. Core will build first and info-storage-service will have dependency of core. So I cant call this method in core which will make it circular dependency. This above method is just for any future usage.